keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+    - `pod.status.hostIP`
+    - `pod.status.hostIPs[].ip`
+    - `pod.status.podIP`
+    - `pod.status.podIPs[].ip`
+    - `service.spec.clusterIP`
+    - `service.spec.clusterIPs[]`
+    - `service.spec.externalIPs[]`
+    - `service.spec.loadBalancerSourceRanges[]`
+    - `service.status.loadBalancer.ingress[].ip`
+
+  - in `networking`:
+    - `ingress.status.loadBalancer.ingress[].ip`
+    - `networkpolicy.spec.egress[].to[].ipBlock.cidr`
+    - `networkpolicy.spec.egress[].to[].ipBlock.except[]`
+    - `networkpolicy.spec.ingress[].from[].ipBlock.cidr`
+    - `networkpolicy.spec.ingress[].from[].ipBlock.except[]`
+    - `servicecidr.spec.cidrs[]`
+
+  - in `discovery`:
+    - `endpointslice.endpoints[].addresses[]`
+
+  - in `resource`:
+    - `resourceclaim.status.devices[].networkData.ips[]`
+
+`servicecidr.spec.cidrs[]` already has stricter semantics than we
+propose here (it requires canonical form for IPv6 values), so while
+its validation function will be updated to make better use of shared
+code, its semantics will not actually change.
+
+`resourceclaim.status.devices[].networkData.ips[]` is part of the
+`DRAResourceClaimDeviceStatus` feature, which is currently alpha, so
+we will also just make it unconditionally require the stricter
+validation. (This field contains "ifaddr"-style CIDR strings, so it
+will either require custom validation or else an additional validation
+helper function used only for that one field.)
+
+`service.spec.loadBalancerSourceRanges[]` has slightly anomalous
+syntax (allowing whitespace around values) for historical reasons. The
+current plan is to require valid CIDR strings to be used, but to not
+be any stricter about whitespace than we are now.
+
+### Canonicalization of Kubernetes-controlled values
+
+Some of the fields above are normally set by Kubernetes-internal code:
+
+  - Set by the Endpoints Controller
+    - `endpoints.subsets[].addresses[].ip`
+    - `endpoints.subsets[].notReadyAddresses[].ip`
+
+  - Set by the EndpointSlice and EndpointSlice Mirroring controllers:
+    - `endpointslice.endpoints[].addresses[]`
+
+  - Set by the NodeIPAM controller:
+    - `node.spec.podCIDR`
+    - `node.spec.podCIDRs[]`
+
+  - Set by kubelet:
+    - `pod.status.hostIP`
+    - `pod.status.hostIPs[].ip`
+    - `pod.status.podIP`
+    - `pod.status.podIPs[].ip`
+
+  - Set by kube-apiserver:
+    - `service.spec.clusterIP`
+    - `service.spec.clusterIPs[]`
+
+  - Set by the cloud-provider framework code:
+    - `service.status.loadBalancer.ingress[].ip`
+
+We will make sure that the code paths that set these values always set
+them to fully-canonical values, even if they initially receive them in
+non-canonical form from some other component.
+
+However, we will still need stricter validation for these fields as
+well, since all of them are either mutable, or else sometimes set by
+external components.
+
+### API Warnings
+
+For several releases before changing the validation rules, the
```

Comment:
I'd explicitly say that since [k8s 1.27](https://github.com/kubernetes/kubernetes/commit/1b804fc87c5586989392116a81bc0d07047dbb87).

Reply:
That was only for the IP/CIDR fields in Service. We're planning to add warnings to all of the other fields in 1.33 (https://github.com/kubernetes/kubernetes/pull/128786).